### PR TITLE
Improvements to Acquiring Lock File

### PR DESCRIPTION
**Pull request by Martinski4GitHub**
Code improvements when the script is trying to obtain a lock file by allowing a 2nd instance to wait for the 1st instance to perform its job and exit. This means that 2 or more instances could execute in a staggered pattern and not terminate immediately when a lock file is found.

### CODE REVIEW

**Comment by ExtremeFiretop** [comment]
@Martinski4GitHub 

Sorry buddy was a busy day and I was tired so I didn't get to test this out.
However I noticed a new alpha for 3006 so I'm going to test this out now and advise! :)

**Review by ExtremeFiretop — APPROVED** [review]

File: MerlinAU.sh [hunk 1]
```diff
@@ -291,38 +291,79 @@ _WaitForYESorNO_()
...
 
+if [ $# -eq 0 ] || [ -z "$1" ]
+then  #Interactive Mode#
+    LockFileMaxAgeSecs=1200
+    LockWaitTimeoutSecs=10
+elif [ "$1" = "addCronJob" ]
+then  #Special Case#
+    LockFileMaxAgeSecs=1200
+    LockWaitTimeoutSecs=600
+fi
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:311]: Ah; I see what you did here; and we can add as many cases as we want for staggered fashion :)

File: MerlinAU.sh [hunk 2]
```diff
@@ -6388,7 +6429,7 @@ _PostUpdateEmailNotification_()
    fi
 
    _UserTraceLog_ "END of $logMsg [$$curWaitDelaySecs sec.]"
-   sleep 60  ## Let's wait a bit & proceed ##
+   sleep 20  ## Let's wait a bit & proceed ##
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:6432]: Approved

File: MerlinAU.sh [hunk 3]
```diff
@@ -6430,7 +6471,7 @@ _PostRebootRunNow_()
    fi
 
    _UserTraceLog_ "END of $logMsg [$$curWaitDelaySecs sec.]"
-   sleep 60  ## Let's wait a bit & proceed ##
+   sleep 30  ## Let's wait a bit & proceed ##
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:6474]: Approved

File: MerlinAU.sh [hunk 4]
```diff
@@ -291,38 +291,79 @@ _WaitForYESorNO_()
...
+          fi
+          sleep 2
+          waitTimeoutSecs="$((waitTimeoutSecs + 2))"
+      else
+          Say "${REDct}**ERROR**${NOct}: The shell script ${ScriptFileName} [PID=$oldPID] is already running [Lock Age: $ageOfLockSecs secs.]"
+          retCode=1
+          break
+      fi
+   done
+   return "$retCode"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:366]: I like the fact that with this; if the lock is active and not stale, the script does not exit immediately. instead it enters a waiting phase where it periodically checks if the lock has been released. :)

**Comment by ExtremeFiretop** [comment]
Tested! And worked perfectly!! :)

image

**Comment by Martinski4GitHub** [comment]
> @Martinski4GitHub
> 
> Sorry buddy was a busy day and I was tired so I didn't get to test this out. However I noticed a new alpha for 3006 so I'm going to test this out now and advise! :)
>
>
No worries, bud. I completely get it. We both have "other things" going on and sometimes we're too tired to do other things.

**Pull request merged**